kibble/models/item_index.go

```diff
@@ -125,14 +144,19 @@ func (itemIndex ItemIndex) findSlugsOfType(slugType string, itemType GenericItem
 	t, ok := itemIndex[slugType]
 	if ok {
 		for k, v := range t {
-			if v == itemType {
+			if deepCompare(v, itemType) {
 				found = append(found, k)
 			}
 		}
 	}
 	return found
 }
 
+//Check fields of GenericItem, ignore StatusID (publish status) as it may differ.
+func deepCompare(a GenericItem, b GenericItem) bool {
```

Comment:
`deepCompare()` sounds like it's recursing properties or something. How about `equalsIgnoreStatus()`?